src/app/UI/lib/managers/auth/index.js

```diff
@@ -72,12 +72,18 @@ class Auth {
 
         const client = ServiceComBus.instance.getClient("userrepo");
         // Get user with email or _id
-        const users = await client.list("user", {
+        let users = await client.list("user", {
             $or: [
                 { _id: emailOrId },
                 { email: emailOrId }
             ]
         });
+        if (users.length === 0) {
+            // If no user is found try to create which will check against other backends for the user
+            // for example an activeDirectory and if password is ok add user to userrepo backend
+            users = [ (await client.create("user", { _id: emailOrId, name: emailOrId, password: password })) ];
```

Comment:
Another solution here could be to us user.push(await ....) then let users above can become const users